File: algorithms/1_easy/S/01876_substrings_of_size_three_with_distinct_characters/solution.cpp
class Solution {
public:
    static int countGoodSubstrings(const string& s) {
        const size_t len = s.size();
        size_t cnt = 0;

        for (size_t i = 2; i < len; i += 1) {
            if (s[i] != s[i - 1] && s[i] != s[i - 2] && s[i - 1] != s[i - 2]) {
                cnt += 1;
            }
        }

        if (cnt > static_cast<size_t>(numeric_limits<int>::max())) { return -1; }

        return static_cast<int>(cnt);
    }
};

File: algorithms/1_easy/S/01876_substrings_of_size_three_with_distinct_characters/solution.rs
impl Solution {
    pub fn count_good_substrings(s: String) -> i32 {
        let mut cnt = 0_usize;

        for w in s.as_bytes().windows(3) {
            if w[0] != w[1] && w[0] != w[2] && w[1] != w[2] {
                cnt += 1;
            }
        }

        cnt.try_into().unwrap()
    }
}
